archive_viewer/table_models/curve_model.py

```diff
@@ -62,26 +64,52 @@ def set_data(self, column_name: str, curve: ArchivePlotCurveItem, value: Any) ->
         """
         ret_code = False
         if column_name == "Channel":
-            if value == curve.address:
-                return True
-
-            [ch.disconnect() for ch in curve.channels() if ch]
-            curve.address = str(value)
-            [ch.connect() for ch in curve.channels() if ch]
+            # If we are changing the channel, then we need to check the current type, and the type we're going to
+            index = self.index(self._plot._curves.index(curve),0)
+            value_is_formula = value.startswith("f://")
+            curve_is_formula = isinstance(curve, FormulaCurveItem)
+            if value_is_formula and not curve_is_formula:
+                #Regardless of starting point, going to a formula is handled in this one function
+                return self.replaceToFormula(index = index, formula = value)
+            elif value_is_formula and curve_is_formula:
+                pvdict = self.formulaToPVDict(index, value)
+                if pvdict:
```

Comment:
If the new formula is a constant (f://5) then the pvdict would be empty and this check would fail.

Something you could try is in `formulaToPVDict()`, instead of printing a message and a `return False`, when it fails you can raise an exception:
```
    if pv not in self._row_names:
        print("Error, " + pv + " is an invalid variable name.")
        return None
```
V  V  V  V  V
```
    if pv not in self._row_names:
        raise ValueError(f"{pv} is an invalid variable name")
```
And then you can replace this if block with a try except block:
```
    pvdict = self.formulaToPVDict(index, value)
    if pvdict:
        curve.formula = value
        curve.pvs = pvdict
    else:
        return False
```
V  V  V  V  V
```
    try:
        pv_dict = self.formulaToPVDict(index, value)
    except ValueError as e:
        logger.error(e)

    curve.formula = value
    curve.pvs = pv_dict
```